src/features/filter.ts

```diff
@@ -1,10 +1,57 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
+type SetStatus = {
+  type: 'filter/SET_STATUS',
+  payload: Status,
+};
+
+const setStatus = (status: Status): SetStatus => {
+  return {
+    type: 'filter/SET_STATUS',
+    payload: status,
+  };
+};
+
+type SetQuery = {
+  type: 'filter/SET_QUERY',
+  payload: string,
+};
+
+const setQuery = (query: string): SetQuery => {
   return {
-    query: '',
-    status: 'all',
+    type: 'filter/SET_QUERY',
+    payload: query,
   };
 };
 
+export const actions = { setStatus, setQuery };
+
+const initialState = {
+  query: '',
+  status: 'all' as Status,
```

Comment:
Avoid using direct type casting (as keyword) if possible. Consider handling the initial state in a more appropriate way or adjust the Status type if necessary.